secondarylib-feedback/src/main/scala/com/socrata/datacoordinator/secondary/feedback/DataCoordinatorClient.scala

```diff
@@ -233,28 +238,35 @@ class HttpDataCoordinatorClient[CT,CV](httpClient: HttpClient,
   implicit val reCodec = SimpleHierarchyCodecBuilder[Response](NoTag).branch[Upsert].branch[NonfatalError].build
 
   override def postMutationScript(script: JArray, cookie: CookieSchema): Option[Either[RequestFailure, UpdateSchemaFailure]] = {
+    val resourceScope = new ResourceScope("post mutation script")
+
     val endpoint = datasetEndpoint.getOrElse(return Some(Left(FailedToDiscoverDataCoordinator)))
     val builder = RequestBuilder(new java.net.URI(endpoint))
 
     def body = JValueEventIterator(script)
 
-    def successHandle(response: JArray): Either[UnexpectedError, Option[UpdateSchemaFailure]] = {
-      assert(response.elems.length == 1, "Response contains more than one element")
-      JsonDecode.fromJValue[JArray](response.elems.head) match {
-        case Right(results) =>
-          assert(results.elems.length == script.elems.length - 2, "Did not get one result for each upsert command that was sent")
-          val rows = script.elems.slice(2, script.elems.length)
-          results.elems.zip(rows).foreach { case (result, row) =>
-            JsonDecode.fromJValue[Response](result) match {
-              case Right(Upsert("update", _, _)) => // yay!
-              case Right(NonfatalError("error", nonfatalError, Some(_))) if nonfatalError == "insert_in_update_only" ||  nonfatalError == "no_such_row_to_update" =>
+    def successHandle(response: Iterator[JValue]): Either[UnexpectedError, Option[UpdateSchemaFailure]] = {
+      try {
+        assert(response.hasNext, "Response contains more than one element")
```

Comment:
You probably want to move this down below the `response.next()` call (right now it's just validating that there is _at least_ one item, not _exactly_ one)